apps/mobile/src/screens/home/home.tsx

```diff
@@ -1,15 +1,70 @@
-import React from 'react'
-
-import { JoinCard } from './components/join-card'
-import { DefaultLayout } from '../../shared/layouts/default-layout'
+import React from "react";
+import { Button, Card, Icon, ThemeProvider } from "ui";
+import { useScanController } from "./hooks/use-scan-controller";
+import { DefaultLayout } from "../../shared/layouts/default-layout";
+import { useVisitData } from "./hooks/use-visit-data";
+import { Text } from 'react-native'
+import { CameraView } from 'expo-camera'
+import { Histories } from "./components/histories";
 
 /**
  * The home screen of the application.
  */
 export function HomeScreen() {
-  return (
+  const { visit, loading } = useVisitData();
+
+  const {
+    allowToScanAgain,
+    handleBarCodeScanned,
+    hasPermissionToUseCamera,
+    isCameraOpen,
+    openCamera,
+    requestPermissionToUseCamera,
+  } = useScanController();
+
+
+  if (isCameraOpen && hasPermissionToUseCamera) {
+    return (
       <DefaultLayout>
-          <JoinCard />
```

Comment:
JoinCard should not be removed. There is no reason to change the Home, except to add your history component